resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/ResourceAddressFactorySpi.java

```diff
@@ -463,5 +471,36 @@ protected void parseNamedOptions0(URI location, ResourceOptions options,
     protected String getRootSchemeName() {
         return null;
     }
+    
+    /**
+     * Throw error on specific circumstances:
+     * - no addresses available for binding
+     * - when PreferedIPv4 flag is true and the host IP is IPV6
+     * @param location
+     */
+    private void throwNoAddressesToBindError(URI location) {
+        StringBuffer error = new StringBuffer(format(NO_ADDRESSES_AVAILABLE_FOR_BINDING, location));
+        try {
+            InetAddress address = InetAddress.getByName(location.getHost());
+            error.insert(0, getErrorMessageIfPreferedIPv4(address));
+        } catch (UnknownHostException e) {
+            // InetAddress.getByName(hostAddress) throws an exception (hostAddress may have an
+            // unsupported format, e.g. network interface syntax)
+        }
+        throw new IllegalArgumentException(error.toString());
+    }
+
+    /**
+     * Verify PreferedIPv4 flag and IPV6 in order to throw a complex error message
```

Comment:
I normally would not comment on this but as there is already things to fix, the first line of JavaDoc should end with punctuation.  Our disabled checkstyle that we want to enable in the future would throw an error on this.